### ISSUE TITLE
evaluations

### ISSUE THREAD

**Issue by priyanga24** [awslabs/autogluon#533]
Hi,
reference from https://arxiv.org/pdf/2003.06505.pdf
![image](https://user-images.githubusercontent.com/34644984/85363714-2c19f380-b554-11ea-856e-648eb4d469e8.png)
For the above datasets, is this the code you used?
```
`from autogluon.task import TabularPrediction as task

predictor = task.fit(train_data=dftrain, label=column_name, hyperparameters={'LR':{'penalty': 'L1'}}, auto_stack=True, eval_metric='r2') 

y_pred = predictor.predict(dftest)`
```

The output gives only the validation score from fit(), not the test data scores.
The test data does not have the ground truth(actual values). We need actual values and the predicted values to calculate metrics like rmse, r2,etc. But since actual values are not available in test.csv then how do we evaluate the model? 

![image](https://user-images.githubusercontent.com/34644984/85365347-bf085d00-b557-11ea-9fb3-ee036df4e154.png)

For example the mercedes benz dataset, from the picture above, we can see that train.csv has 'y' column, while test.csv doesn't have 'y' column. 
In the below line, we don't have y_test(actual values) to evaluate our model.
`predictor.evaluate_predictions(y_true=y_test, y_pred=y_pred)`
Where can we obtain the true values? Thanks!

**Comment by Innixma** [comment]
You have to join the Kaggle competition and submit your predictions to Kaggle to get the test score, the test labels are not publicly available.

**Comment by priyanga24** [comment]
ok! what was the algorithm you used? hyperparams={'LR':{'penalty':'L1'}} or all the algorithms? 
And using L1 loss gave bad results for 'house price' dataset. I see that it's mentioned in the paper that you've used L1 loss.

**Comment by Innixma** [comment]
Please refer to the AutoGluon-Tabular paper for extensive discussion of the experimental setup: https://arxiv.org/pdf/2003.06505.pdf

We used default hyperparameters, without them being manually set. We did not use linear regression.

For reproducing results exactly: https://github.com/Innixma/autogluon-benchmarking/blob/master/autogluon_utils/benchmarking/kaggle/README-kagglebenchmark.txt

For exact eval metric used, see https://github.com/Innixma/autogluon-benchmarking/blob/master/autogluon_utils/configs/kaggle/kaggle_competitions.py

For exact parameters used by AutoGluon, see: https://github.com/Innixma/autogluon-benchmarking/blob/master/autogluon_utils/configs/kaggle/kaggle_competitions.py#L195

Note this was using autogluon==0.0.5, and therefore is somewhat outdated.

For understanding what arguments do such as auto_stack: https://autogluon.mxnet.io/api/autogluon.task.html#tabularprediction

**Comment by priyanga24** [comment]
Thanks for your help!
I was able to experiment it, and for this dataset https://www.kaggle.com/c/santander-customer-satisfaction/data , I see that your roc_auc values was 0.82654. But my score was only 0.50065.
I trained for the model for 4 hours as you did, and set auto_stack=True. 
Now Because you've upgraded autogluon , due to some changes my score is less now compared to the score with the version you used while doing the experiment? or what could be the reason? Please help, thanks

**Comment by Innixma** [comment]
0.5 with AUC is the score of randomly guessing. You are feeding in predictions (1 and 0) when you should be submitting prediction probabilities (0.8, 0.2, etc.). Prediction probabilities are required to calculate AUC, otherwise you will see a score very close to 0.5.

**Comment by priyanga24** [comment]
oh Thanks! Got confused with the sample submission. Sorry, thanks for the help!
